e2e/teams.loggedin.spec.ts

```diff
@@ -0,0 +1,195 @@
+      ).toBeVisible()
+
+      await userPage.getByRole('button', { name: 'Create a New Team' }).click()
+      await userPage.getByRole('textbox').nth(1).fill(TEAM_NAME)
+      await userPage.getByRole('button', { name: 'OK' }).click()
+      await expect(
+        userPage.getByText(`Team Created: ${TEAM_NAME}`),
+      ).toBeVisible()
+    })
+
+    await test.step('user should invite admin to team', async () => {
+      await userPage.getByRole('button', { name: 'Team Actions' }).click()
+      await userPage.getByRole('link', { name: 'Invite User' }).click()
+
+      await userPage
+        .getByRole('textbox', { name: 'Enter a name or email address...' })
+        .type(adminUserName!)
+      await userPage
+        .getByRole('menuitem', { name: `@${adminUserName}` })
+        .click()
+      const loadInvitedUser = userPage.getByText('Loading...')
+      await expect(loadInvitedUser).toBeVisible()
+      await expect(loadInvitedUser).not.toBeVisible()
+
+      await userPage.getByLabel('Invitation Message').type(INVITATION_MESSAGE)
+
+      const spinner = userPage.locator('.modal-body > .spinner')
+      await Promise.all([
+        expect(spinner).toBeVisible(),
+        userPage.getByRole('button', { name: 'Send Invitation(s)' }).click(),
+      ])
+
+      await expect(spinner).not.toBeVisible()
+      await expect(userPage.getByText('Invitation(s) Sent')).toBeVisible()
+    })
+
+    await test.step('user should view pending invitations', async () => {
+      await expect(userPage.getByText('Pending Invitations')).toBeVisible({
+        timeout: testInfo.timeout * 3, // add extra timeout, so backend can finish sending request
+      })
+      const row = userPage.getByRole('row', { name: adminUserName! })
+      await expect(row).toBeVisible()
+      await expect(row.getByText(INVITATION_MESSAGE)).toBeVisible()
+    })
+
+    await test.step('admin should accept team invitation', async () => {
+      await goToDashboard(adminPage)
+
+      await Promise.all([
+        expect(
+          adminPage.getByRole('heading', { name: 'Your Teams' }),
+        ).toBeVisible(),
+        adminPage.getByLabel('Teams').click(),
+      ])
+
+      // get row for this invitation
+      // ...in case multiple test suite users have invited admin user at the same time
+      const row = adminPage.getByRole('row', { name: TEAM_NAME })
+      await expect(row.getByText(INVITATION_MESSAGE)).toBeVisible()
+
+      await row.getByRole('button', { name: 'Join' }).click()
+      await expect(row).not.toBeVisible()
+    })
+
+    await test.step('admin should view team page', async () => {
+      const teamLink = adminPage.getByRole('link', { name: TEAM_NAME })
+
+      // handle case where occasionally two team links are shown
+      // ...before resolving to one team link
+      await expect
+        .poll(async () => {
+          return await teamLink.count()
+        })
+        .toBe(1)
+
+      await teamLink.click()
```

Comment:
Playwright locators are [strict](https://playwright.dev/docs/locators#strictness), so an exception will be thrown if we try to click on a locator that matches more than one element. Poll will wait for the link to resolve to a single element before trying to click on it.